### PR TITLE
Titus control plane changes to add new GRPC service as a proxy for VpcService

### PR DESCRIPTION
**Pull request by MyUmmaGumma**
This change creates a proxy GRPC service in Titus control plane under federation, as a front end to VpcService with the intention of allowing Spinnaker to call this metatron enabled proxy and in turn relays the call using plaintext to VpcService.

### CODE REVIEW

File: titus-server-federation/src/main/java/com/netflix/titus/federation/endpoint/grpc/FederationGrpcComponent.java [hunk 1]
```diff
@@ -16,11 +16,7 @@
 
 package com.netflix.titus.federation.endpoint.grpc;
 
-import com.netflix.titus.runtime.endpoint.v3.grpc.DefaultAutoScalingServiceGrpc;
-import com.netflix.titus.runtime.endpoint.v3.grpc.DefaultHealthServiceGrpc;
-import com.netflix.titus.runtime.endpoint.v3.grpc.DefaultJobActivityHistoryServiceGrpc;
-import com.netflix.titus.runtime.endpoint.v3.grpc.DefaultJobManagementServiceGrpc;
-import com.netflix.titus.runtime.endpoint.v3.grpc.DefaultLoadBalancerServiceGrpc;
+import com.netflix.titus.runtime.endpoint.v3.grpc.*;
```
Comment (tbak) [inline titus-server-federation/src/main/java/com/netflix/titus/federation/endpoint/grpc/FederationGrpcComponent.java:19]: Please, do not use wildcards in imports

File: titus-server-federation/src/main/java/com/netflix/titus/federation/service/DefaultVpcServiceConnector.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    private static final long SHUTDOWN_TIMEOUT_MS = 5_000;
+    private final TitusFederationConfiguration configuration;
+    private final ManagedChannel channel;
+
+    @Inject
+    public DefaultVpcServiceConnector(TitusFederationConfiguration configuration) {
+        this.configuration = configuration;
+        String []hostPort = configuration.getVpcService().split(":");
+
+        channel = NettyChannelBuilder
```
Comment (tbak) [inline titus-server-federation/src/main/java/com/netflix/titus/federation/service/DefaultVpcServiceConnector.java:22]: We should not use plain text in the production deployment.

File: titus-server-runtime/src/main/java/com/netflix/titus/runtime/service/TitusAgentSecurityGroupClient.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.netflix.titus.runtime.service;
+
+import com.netflix.titus.api.model.callmetadata.CallMetadata;
+import com.netflix.titus.TitusVpcApi.ResetSecurityGroupRequest;
+import com.netflix.titus.TitusVpcApi.ResetSecurityGroupResponse;
+import rx.Observable;
+
+public interface TitusAgentSecurityGroupClient {
+    Observable<ResetSecurityGroupResponse> ResetSecurityGroup(ResetSecurityGroupRequest request, CallMetadata callMetadata);
```
Comment (tbak) [inline titus-server-runtime/src/main/java/com/netflix/titus/runtime/service/TitusAgentSecurityGroupClient.java:9]: Please, use Spring Reactor instead. This is legacy RX library.

File: titus-server-federation/src/main/java/com/netflix/titus/federation/service/DefaultTitusAgentSecurityGroupClient.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Inject
+    public DefaultTitusAgentSecurityGroupClient(VpcServiceConnector connector,
+                                                GrpcConfiguration grpcConfiguration) {
+        this.clientToVpcService = TitusAgentSecurityGroupServiceGrpc.newStub(connector.getChannel());
+        this.grpcConfiguration = grpcConfiguration;
+    }
+
+    @Override
+    public Observable<TitusVpcApi.ResetSecurityGroupResponse> ResetSecurityGroup(
```
Comment (tbak) [inline titus-server-federation/src/main/java/com/netflix/titus/federation/service/DefaultTitusAgentSecurityGroupClient.java:31]: Methods should start with a smaller case letter.

File: titus-server-federation/src/main/java/com/netflix/titus/federation/service/DefaultTitusAgentSecurityGroupClient.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            TitusVpcApi.ResetSecurityGroupRequest request, CallMetadata callMetadata) {
+        return createRequestObservable(emitter -> {
+            StreamObserver<ResetSecurityGroupResponse> streamObserver;
+            streamObserver = GrpcUtil.createClientResponseObserver(
```
Comment (tbak) [inline titus-server-federation/src/main/java/com/netflix/titus/federation/service/DefaultTitusAgentSecurityGroupClient.java:35]: Do not use these helper. It should be marked as legacy. Instead bridge directly GRPC stream to Reactor Mono.

**Review by tbak — CHANGES_REQUESTED** [review]

File: titus-server-federation/src/main/java/com/netflix/titus/federation/service/DefaultTitusAgentSecurityGroupClient.java (hunk 5 above)
Comment (MyUmmaGumma) [reply titus-server-federation/src/main/java/com/netflix/titus/federation/service/DefaultTitusAgentSecurityGroupClient.java]: Done.

File: titus-server-federation/src/main/java/com/netflix/titus/federation/service/DefaultTitusAgentSecurityGroupClient.java (hunk 4 above)
Comment (MyUmmaGumma) [reply titus-server-federation/src/main/java/com/netflix/titus/federation/service/DefaultTitusAgentSecurityGroupClient.java]: Done.

File: titus-server-federation/src/main/java/com/netflix/titus/federation/service/DefaultVpcServiceConnector.java (hunk 2 above)
Comment (MyUmmaGumma) [reply titus-server-federation/src/main/java/com/netflix/titus/federation/service/DefaultVpcServiceConnector.java:22]: VpcService does not support metatron, which is the reason for this proxy via the control-plane.

File: titus-server-federation/src/main/java/com/netflix/titus/federation/endpoint/grpc/FederationGrpcComponent.java (hunk 1 above)
Comment (MyUmmaGumma) [reply titus-server-federation/src/main/java/com/netflix/titus/federation/endpoint/grpc/FederationGrpcComponent.java]: Done.

**Review by tbak — APPROVED** [review]

**Review by tbak — APPROVED** [review]

**Pull request merged**